packages/macro/src/macroJs.ts

```diff
@@ -148,13 +151,62 @@ export default class MacroJs {
       this.isLinguiIdentifier(path.node.callee, "t")
     ) {
       this.replaceTAsFunction(path as NodePath<CallExpression>)
+      this.needsI18nImport = true
+
       return true
     }
 
+    // { t } = useLingui()
```

Comment:
I'm sorry, I don't really follow these comments (and the conditions neither)

if the comment were a little longer that'd be nice.

Reply:
It's easier to follow the logic when you read whole file not only diff, there is a list of possible syntaxes:

![image](https://github.com/lingui/js-lingui/assets/1586852/50782331-ef08-4214-a4ff-cae5fa245d4f)

This comments is extensions to this list. Purpose of this comment is to find an entry point of specific syntax to transform. 

I could agree that the code is not very much readable, but this is how usually transforms looks like. They a very wordy and not readable, there is not much we can do about it.